http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/ResponseHandler.java

```diff
@@ -463,18 +464,28 @@ public void cancel() {
     private void notifyIfResponseNotCompleted(ChannelHandlerContext handlerCtx) {
         RequestContext requestCtx = handlerCtx.channel().attr(REQUEST_CONTEXT_KEY).get();
         Boolean responseCompleted = handlerCtx.channel().attr(RESPONSE_COMPLETE_KEY).get();
-        Boolean lastHttpContentReceived = handlerCtx.channel().attr(LAST_HTTP_CONTENT_RECEIVED_KEY).get();
+        boolean isLastByteWithout100Continue = isLastByteWithout100Response(handlerCtx);
         handlerCtx.channel().attr(KEEP_ALIVE).set(false);
 
-        if (!Boolean.TRUE.equals(responseCompleted) && !Boolean.TRUE.equals(lastHttpContentReceived)) {
+        if (!Boolean.TRUE.equals(responseCompleted) && !isLastByteWithout100Continue) {
             IOException err = new IOException(NettyUtils.closedChannelMessage(handlerCtx.channel()));
             runAndLogError(handlerCtx.channel(), () -> "Fail to execute SdkAsyncHttpResponseHandler#onError",
                            () -> requestCtx.handler().onError(err));
             executeFuture(handlerCtx).completeExceptionally(err);
             runAndLogError(handlerCtx.channel(), () -> "Could not release channel", () -> closeAndRelease(handlerCtx));
+        } else if (!Boolean.TRUE.equals(responseCompleted)) {
```

Comment:
The logic is a bit confusing. Does it make sense to rearrange it with `!Boolean.TRUE.equals(responseCompleted)` in an outer if clause and then just a simple if/else checking `isLastByteWithout100Response` nested within?
~~~
if (!Boolean.TRUE.equals(responseCompleted)) {
    if (!isLastByteWithout100Continue) {
            IOException err = new IOException(NettyUtils.closedChannelMessage(handlerCtx.channel()));
            runAndLogError(handlerCtx.channel(), () -> "Fail to execute SdkAsyncHttpResponseHandler#onError",
                           () -> requestCtx.handler().onError(err));
            executeFuture(handlerCtx).completeExceptionally(err);
            runAndLogError(handlerCtx.channel(), () -> "Could not release channel", () -> closeAndRelease(handlerCtx));
        } else {
            log.trace(handlerCtx.channel(),
                      () -> "Run error skipped because lastHttpContentReceived is "
                            + handlerCtx.channel().attr(LAST_HTTP_CONTENT_RECEIVED_KEY).get() + " and 100ContinueMessage is "
                            + handlerCtx.channel().attr(RESPONSE_100_CONTINUE_MESSAGE).get());
        }
}

Reply:
not required in new changes